hyper-boring/src/lib.rs

```diff
@@ -11,12 +11,21 @@ use std::fmt;
 use tokio_boring::SslStream;
 
 mod cache;
+#[cfg(all(feature = "hyper0", not(feature = "hyper1")))]
 mod v0;
+/// Hyper 0 support.
+#[cfg(all(feature = "hyper0", feature = "hyper1"))]
+pub mod v0;
+#[cfg(all(feature = "hyper1", not(feature = "hyper0")))]
+mod v1;
 /// Hyper 1 support.
-#[cfg(feature = "hyper1")]
+#[cfg(all(feature = "hyper1", feature = "hyper0"))]
 pub mod v1;
 
+#[cfg(all(feature = "hyper0", not(feature = "hyper1")))]
 pub use self::v0::*;
+#[cfg(all(feature = "hyper1", not(feature = "hyper0")))]
+pub use self::v1::*;
```

Comment:
This means the features aren't purely additive, and could create confusing type errors if both flags end up being set by accident. Maybe always keep `v0` in a module?

Reply:
> This means the features aren't purely additive, and could create confusing type errors if both flags end up being set by accident.

To be clear, the objective of the `#[cfg]` gate is to properly take that case into consideration: if both are set, then the items are kept in explicit module paths, but if only one is set, all items are re-exported in order to keep a simpler pathing in what should be most cases I imagine. This should help avoid type collisions like you mentioned. Couldn't this work you think?

> Maybe always keep `v0` in a module?

I can do that, but it would mean breaking API even more: current paths are `crate::HttpsConnector` for the v0 and `crate::v1::HttpsConnector` for the v1, while inverting like you suggest would do `crate::v0::HttpsConnector` for the v0 and crate::HttpsConnector` for the v1, I guess. Is this still acceptable?

Reply:
The names of features have changed, so this will need to be a semver-major release anyway.

I am concerned about the items re-exported at the root, because they change in non-additive way. The feature flags are okay if users only used `v0::…`/`v1::…` types, but the re-export can be a problem if users mix use of the top-level items with `v0::…` items, or top-level items with another crate.

Re-exporting only v1 types at the top level would be more robust.

Reply:
Alright by me! I was just trying not to break too much if possible.

Also, on top of that and since it seems we go more in the direction of v1, I've made it so v1 is now the default and v0 is not anymore but still available through the same features.